src/main/java/net/minecraftforge/event/world/BlockEvent.java

```diff
@@ -444,4 +444,47 @@ public PortalSpawnEvent(IWorld world, BlockPos pos, BlockState state, NetherPort
             return size;
         }
     }
+    
+    /**
+     * Fired when when this block is right clicked by a tool to change its state.
+     * Used to determine if an axe can strip, a shovel can path, or a hoe can till.
+     * 
+     * This event is {@link Cancelable}. If canceled, this will prevent the tool
+     * from changing the block's state.
+     */
+    @Cancelable
+    public static class BlockToolInteractEvent extends BlockEvent
+    {
+
+    	private final PlayerEntity player;
+    	private final ItemStack stack;
+    	private BlockState finalState;
```

Comment:
Should probably just be state. (Nitpicky)

Reply:
Referenced as finalState to know the diference between getState and getFinalState in code. However, I can change to state. (less nitpicky)

Reply:
You can resolve this, it's valid since there is already a getState, setState, and original state things.